File: backend/app/models/usage_log.py
from sqlalchemy import Column, Integer, String, DateTime, func
from app.database import Base


class UsageLog(Base):
    __tablename__ = "usage_logs"

    id              = Column(Integer, primary_key=True, index=True)
    user_id         = Column(Integer, nullable=False, index=True)
    statement_id    = Column(Integer, nullable=True)
    pages_consumed  = Column(Integer, nullable=False, default=1)
    action          = Column(String, nullable=False, default="upload")  # upload|manual
    created_at      = Column(DateTime, default=func.now())

File: backend-java/src/main/java/com/expensetracker/model/UsageLog.java
package com.expensetracker.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.OffsetDateTime;

@Entity
@Table(name = "usage_logs")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class UsageLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @EqualsAndHashCode.Include
    private Long id;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(name = "statement_id")
    private Long statementId;

    @Column(name = "pages_consumed", nullable = false)
    @Builder.Default
    private int pagesConsumed = 1;

    @Column(nullable = false)
    @Builder.Default
    private String action = "upload";

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt;
}
